test/index.js

```diff
@@ -1117,6 +1118,63 @@ describe('list', function() {
 
   });
 
+  describe('slice', function() {
+
+    it('returns a Nothing with a positive end index greater than start index', function() {
+      assert(S.slice(6, 1, [1, 2, 3, 4, 5]).equals(S.Nothing()));
+    });
+
+    it('returns a Nothing with a positive end index greater than list length', function() {
+      assert(S.slice(1, 6, [1, 2, 3, 4, 5]).equals(S.Nothing()));
+    });
+
+    it('returns a Nothing with a negative end index greater than list length', function() {
+      assert(S.slice(1, -6, [1, 2, 3, 4, 5]).equals(S.Nothing()));
+    });
+
+    it('returns a Nothing with a negative start index greater than list length', function() {
+      assert(S.slice(-6, 1, [1, 2, 3, 4, 5]).equals(S.Nothing()));
+    });
+
+    it('returns a Just with an empty array when start index equals end index', function() {
+      eq(S.slice(1, 1, [1, 2, 3, 4, 5]).toString(), 'Just([])');
+      eq(S.slice(1, -4, [1, 2, 3, 4, 5]).toString(), 'Just([])');
+      eq(S.slice(-4, 1, [1, 2, 3, 4, 5]).toString(), 'Just([])');
+      eq(S.slice(-4, -4, [1, 2, 3, 4, 5]).toString(), 'Just([])');
+      eq(S.slice(0, 0, []).toString(), 'Just([])');
+      eq(S.slice(0, -0, []).toString(), 'Just([])');
+      eq(S.slice(-0, 0, []).toString(), 'Just([])');
+      eq(S.slice(-0, -0, []).toString(), 'Just([])');
+    });
+
+    it('returns a Just with a positive start and end index', function() {
+      eq(S.slice(1, 3, [1, 2, 3, 4, 5]).toString(), 'Just([2, 3])');
+    });
+
+    it('returns a Just with a negative start index', function() {
+      eq(S.slice(-3, 5, [1, 2, 3, 4, 5]).toString(), 'Just([3, 4, 5])');
+    });
+
+    it('returns a Just with a negative end index', function() {
+      eq(S.slice(1, -2, [1, 2, 3, 4, 5]).toString(), 'Just([2, 3])');
+    });
+
+    it('accepts -0 for a start index as the position half a step beyond the last index',
+      function() {
+        eq(S.slice(2, -0, [1, 2, 3, 4, 5]).toString(), 'Just([3, 4, 5])');
+    });
```

Comment:
It should indeed, I mixed the body of the two tests up there. I've actually rolled them into one test block now.